app/main/views/drafts.py

```diff
@@ -418,10 +418,9 @@ def publish_draft_service(draft_id):
         db.session.rollback()
         action = 'update' if draft.status == 'submitted' else 'delete'
         current_app.logger.warning(
-            'Failed to {action} draft {draft_id} after publishing service {service_id}: {error}'.format(
-                extra=dict(
-                    action=action, draft_id=draft_id, service_id=service_from_draft.service_id, error=e.message)))
-
+            f'Failed to {action} draft {draft_id} '
+            f'after publishing service {service_from_draft.service_id}: {e.statement}'
```

Comment:
Just spotted this - why `e.statement` instead of `e.message`?

Reply:
As far as I can tell [from the docs](https://docs.sqlalchemy.org/en/13/core/exceptions.html#sqlalchemy.exc.IntegrityError) that's how SQLAlchemy has written the class 🤔

Reply:
I think we can just pass `e` if the string is going to be formatted anyway - we do this in `views.suppliers`

Reply:
Done!